### PR TITLE
Add villager expressions

### PR DESCRIPTION
**Pull request by ShaneBeee**
### Description
This PR aims to add some useful expressions for villagers:
- Villager Level
- Villager Profession
- Villager Type

### NOTES:
- I've added an ExprBiome.sk test to make sure there were no conflicts with biomes.
- Villager Profession Lang entries: Due to clashing with EntityData and Skript saying `can't compare a villager profession with a living entity` I had to add "profession" to each lang entry. I don't love this, but I don't see any way around it other than changing Villager's EntityData entries in the lang file.
- Enum/Registry ClassInfo changes are from #7251 but were needed for this PR
- ~~MathUtils was added due to `Math.clamp()` not being available on Java 17 (was added in Java 21)~~

---
**Target Minecraft Versions:** any 
**Requirements:** Relies on #7251
**Related Issues:** none

### CODE REVIEW

**Comment by ShaneBeee** [comment]
Closing this for now.
Too many errors on Java 17 as well as old enum class that I forgot changed to interfaces.

**Review by TheAbsolutionism — APPROVED** [review]

File: src/main/resources/lang/default.lang [hunk 1]
```diff
@@ -2413,6 +2413,34 @@ item display transforms:
...
+	toolsmith: toolsmith, toolsmith profession
+	librarian: librarian, librarian profession
+	shepherd: shepherd, shepherd profession
+	farmer: farmer, farmer profession
+	cleric: cleric, cleric profession
+	nitwit: nitwit, nitwit profession
+	cartographer: cartographer, cartographer profession
+	armorer: armorer, armorer profession
+	butcher: butcher, butcher profession
+	none: none, none profession, no profession, unemployed
```
Comment (sovdeeth) [inline src/main/resources/lang/default.lang:2441]: worried about none being an option and therefore ``if {_X} is none`` being valid syntax.
I would rather see this be just the latter two options.

File: src/test/skript/tests/syntaxes/expressions/ExprVillagerType.sk [hunk 2]
```diff
@@ -0,0 +1,11 @@
+test "villager type expression":
+	spawn a villager at event-location:
+		set {_e} to event-entity
+
+	set villager type of {_e} to plains
+	assert villager type of {_e} = plains with "The villager should now have the type plains"
+	set villager type of {_e} to desert
+	assert villager type of {_e} = desert with "The villager should now have the type desert"
+
+	# Thank you for your service
```
Comment (sovdeeth) [inline src/test/skript/tests/syntaxes/expressions/ExprVillagerType.sk:10]: 🫡

**Review by sovdeeth — CHANGES_REQUESTED** [review]

File: src/main/java/ch/njol/skript/util/MathUtils.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+	 * @param max   Max value for clamping
+	 * @return Value, modifier if outside of clamped values
+	 */
+	public static int clamp(final int value, final int min, final int max) {
+		if (min > max)
+			throw new IllegalArgumentException(min + " > " + max);
+		return Math.max(min, Math.min(max, value));
+	}
+
+}
```
[the remark this answers is not in the document]
Comment (ShaneBeee) [reply src/main/java/ch/njol/skript/util/MathUtils.java:26]: Ahh, good to know, I will remove this.

File: src/main/java/ch/njol/skript/expressions/ExprVillagerLevel.java [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+	public @Nullable Number convert(LivingEntity from) {
+		if (from instanceof Villager villager)
+			return villager.getVillagerLevel();
+		return null;
+	}
+
+	@Override
+	public Class<?> @Nullable [] acceptChange(ChangeMode mode) {
+		return switch (mode) {
+			case SET, ADD, REMOVE, DELETE, RESET ->
```
[the remark this answers is not in the document]
Comment (ShaneBeee) [reply src/main/java/ch/njol/skript/expressions/ExprVillagerLevel.java:48]: Thats fair yeah.

File: src/main/resources/lang/default.lang (hunk 1 above)
Comment (ShaneBeee) [reply src/main/resources/lang/default.lang:2454]: Good call, I will remove that.

**Review by sovdeeth — APPROVED** [review]

File: src/main/java/ch/njol/skript/expressions/ExprVillagerLevel.java [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+import ch.njol.util.Math2;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.entity.Villager;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+
+@Name("Villager Level")
+@Description({"Represents the level of a villager.",
+	"Level must be between 1 and 5, with 1 being the default level.",
```
Comment (Efnilite) [inline src/main/java/ch/njol/skript/expressions/ExprVillagerLevel.java:19]:
```suggestion
@Description({
	"Represents the level of a villager.",
	"Level must be between 1 and 5, with 1 being the default level.",
```

File: src/main/java/ch/njol/skript/expressions/ExprVillagerType.java [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.entity.Villager;
+import org.bukkit.entity.Villager.Type;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+
+@Name("Villager Type")
+@Description("Represents the Type of a villager.")
```
Comment (Efnilite) [inline src/main/java/ch/njol/skript/expressions/ExprVillagerType.java:17]:
```suggestion
@Description("Represents the type of a villager. This usually represents the biome the villager is from.")
```
just for clarity. on first read, i didnt know what this signified

File: src/main/java/ch/njol/skript/classes/data/BukkitClasses.java [hunk 7]
```diff
@@ -1565,6 +1566,19 @@ public String toVariableNameString(EnchantmentOffer eo) {
 			.name("Experience Cooldown Change Reason")
 			.description("Represents a change reason of an <a href='events.html#experience cooldown change event'>experience cooldown change event</a>.")
 			.since("INSERT VERSION"));
+
+		Classes.registerClass(new RegistryClassInfo<>(Villager.Type.class, Registry.VILLAGER_TYPE, "villagertype", "villager types")
+			.user("villager ?types?")
+			.name("Villager Type")
+			.description("Represents the different types of villagers.")
```
Comment (Efnilite) [inline src/main/java/ch/njol/skript/classes/data/BukkitClasses.java:1573]:
```suggestion
			.description("Represents the different types of villagers. These are usually the biomes a villager can be from.")
```

**Review by Efnilite — CHANGES_REQUESTED** [review]

**Review by Efnilite — APPROVED** [review]

**Pull request merged**